lib/BaseComponent.js

```diff
@@ -65,6 +67,13 @@ export default inherit(Component, {
 
     getChildContext() {
         if(!this.elem) return { bemBlock : this.block };
+    },
+
+    generateId(key = '__uniqId') {
+        this.__uniqId = this.__uniqId || {};
+        return this.__uniqId[key]
+            ? this.__uniqId[key]
```

Comment:
that's against current code style practice https://github.com/bem/bem-react-core/blob/master/lib/decls.js#L30 ;-) let's add lint check